drivers/ethernet/eth_stellaris.c

```diff
@@ -244,8 +242,9 @@ void eth_stellaris_rx(struct device *dev)
 
 	ret = net_recv_data(iface, pkt);
 	if (ret < 0) {
-		eth_rx_error_out("Failed to place frame in RX Queue", iface);
 		net_pkt_unref(pkt);
+		SYS_LOG_ERR("Failed to place frame in RX Queue");
+		eth_rx_error_out(iface);
```

Comment:
Suggest that we re-organize code as shown below:

``````
void eth_stellaris_rx(struct device *dev)
{
	pkt = net_pkt_get_reserve_rx(...);
	if (!pkt) {
		SYS_LOG_ERR("Could not allocate pkt");
		goto err_mem;
	}
	
	ret = eth_stellaris_rx_pkt(dev, pkt);
	if (ret == -1)
		goto err_rx;

	return;

err_rx:
	net_pkt_unref(pkt);
	
err_mem:
	eth_stellarix_error_out(...);
}
``````